subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -625,6 +626,87 @@ static int cmd_cache(const struct shell *sh, size_t argc, char **argv)
 #endif
 }
 
+static void shell_print_attr(const struct shell *sh, void *ref)
+{
+	struct lwm2m_attr *attr = NULL;
+	bool found;
+	for (uint8_t type = 0; type < NR_LWM2M_ATTR; type++) {
+		found = false;
+		while ((attr = lwm2m_engine_get_next_attr(ref, attr)) != NULL) {
+			if (attr->type == type) {
+				found = true;
+				break;
+			}
+		}
+		if (found) {
+			switch (type) {
+			case LWM2M_ATTR_PMIN:
+			case LWM2M_ATTR_PMAX:
+				shell_fprintf(sh, SHELL_NORMAL, "%10u", attr->int_val);
+				break;
+			case LWM2M_ATTR_GT:
+			case LWM2M_ATTR_LT:
+			case LWM2M_ATTR_STEP:
+				shell_fprintf(sh, SHELL_NORMAL, "%10f", attr->float_val);
+				break;
+			}
+		} else {
+			shell_fprintf(sh, SHELL_NORMAL, "%10s", "");
+		}
+	}
+}
+
+static int cmd_observations(const struct shell *sh, size_t argc, char **argv)
+{
+	char buf[LWM2M_MAX_PATH_STR_SIZE];
+	struct lwm2m_obj_path_list *o_p;
+	struct observe_node *obs;
+	uint32_t i = 0, path_i;
+	struct lwm2m_ctx *ctx;
+	void *ref;
+	int ret;
+
+	ctx = lwm2m_rd_client_ctx();
+	if (ctx == NULL) {
+		shell_error(sh, "no lwm2m context yet\n");
+		return -ENOEXEC;
+	}
+
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	shell_fprintf(sh, SHELL_INFO, " # %10s%10s", "composite", "path");
+	for (i = 0; i < NR_LWM2M_ATTR; i++) {
+		shell_fprintf(sh, SHELL_INFO, "%10s", lwm2m_attr_to_str(i));
+	}
+	shell_print(sh, "");
+
+	lwm2m_registry_lock();
+	i = 0;
+	SYS_SLIST_FOR_EACH_CONTAINER(&ctx->observer, obs, node) {
+		shell_fprintf(sh, SHELL_NORMAL, "%2u %10c", i, obs->composite ? 'y' : 'n');
```

Comment:
```suggestion
		shell_fprintf(sh, SHELL_NORMAL, "%2u %10c ", i, obs->composite ? 'y' : 'n');
```

Add space after y/n character, otherwise long paths will start without the space.

Reply:
Done